File: src/types/attachment.ts
export interface Attachment {
  id: string;
  mail_id: string;
  filename: string;
  mime_type: string;
  size: number | null;
  file_path: string | null;
  content_id: string | null;
}

/** 本文中の `<img src="cid:...">` に対応する画像データ */
export interface InlineImage {
  content_id: string;
  data_uri: string;
}

File: src-tauri/src/models/attachment.rs
use serde::{Deserialize, Serialize};

/// メールの添付ファイル。実体はローカルキャッシュ（file_path）に保存される。
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Attachment {
    pub id: String,
    pub mail_id: String,
    pub filename: String,
    pub mime_type: String,
    pub size: Option<i64>,
    /// キャッシュファイルの絶対パス。ファイルが消えている場合はキャッシュミス扱い
    pub file_path: Option<String>,
}
